spec/monotonic_tick_count_spec.rb

```diff
@@ -0,0 +1,125 @@
+require "spec_helper"
+
+describe MonotonicTickCount do
+  it "should have a version number" do
+    expect(::MonotonicTickCount::VERSION).to be
+  end
+
+  context "#initialize" do
+    it "should snapshot the current tick counter when constructed with no arguments" do
+      expect(Process).to receive(:clock_gettime).with(Process::CLOCK_MONOTONIC) { 1234.012345 }
+      tick_count = MonotonicTickCount.new
+      expect(tick_count.tick_count_f).to eq(1234.012345)
+    end
+
+    it "should have a copy constructor" do
+      tick_count1 = MonotonicTickCount.new
+      tick_count2 = MonotonicTickCount.new(tick_count1)
+      expect(tick_count2.tick_count_f).to eq(tick_count1.tick_count_f)
+    end
+
+    it "should raise an exception from copy constructor if rhs isn't same class or equivalent" do
+      expect { MonotonicTickCount.new(1122.33) }.to raise_error(ArgumentError, "Must initialize from MonotonicTickCount or equivalent")
+    end
+
+    it "should be constructable from a raw tick_count_f" do
+      tick_count = MonotonicTickCount.new(tick_count_f: 112233.4455)
+      expect(tick_count.tick_count_f).to eq(112233.4455)
+    end
+  end
+
+  context "#-" do
+    it "should return the difference in seconds" do
+      tick_count1 = MonotonicTickCount.new(tick_count_f: 123.1)
+      tick_count2 = MonotonicTickCount.new(tick_count_f: 125.6)
+      difference = tick_count2 - tick_count1
+      expect(difference).to eq(2.5.seconds)
+    end
+
+    it "should raise an exception if rhs isn't the same class or equivalent" do
+      tick_count1 = 123.1
+      tick_count2 = MonotonicTickCount.new(tick_count_f: 125.6)
+      expect { difference = tick_count2 - tick_count1 }.to raise_error(ArgumentError, "Other operand must be a MonotonicTickCount or equivalent")
+    end
+  end
+
+  context "#+" do
+    it "should support adding a duration" do
+      tick_count = MonotonicTickCount.new(tick_count_f: 123.1)
+      duration = 2.5.seconds
+      sum = tick_count + duration
+      expect(sum.tick_count_f).to eq(125.6)
+    end
+
+    it "should raise an exception if rhs isn't an ActiveSupport::Duration or equivalent" do
+      tick_count = MonotonicTickCount.new(tick_count_f: 123.1)
+      expect { tick_count + 2.5 }.to raise_error(ArgumentError, "Other operand must be an ActiveSupport::Duration or equivalent")
+    end
+  end
+
+  context "Comparable" do
```

Comment:
Should there be a test in this context for the duck type check in `#<=>`?

Reply:
Yep, added.